confd/pkg/backends/calico/client.go

```diff
@@ -899,8 +892,22 @@ func (c *client) onUpdates(updates []api.Update, needUpdatePeersV1 bool) {
 	// If configuration relevant to BGP peerings has changed, recalculate the set of v1
 	// peerings that should exist, and update the cache accordingly.
 	if needUpdatePeersV1 {
+		// Mark currently watched secrets as stale, so that they can be cleaned up if no
```

Comment:
Hm, why did these need to move?

Reply:
Since the update on the BGP config is only for the password which is dependent on the secret cache, if the secret cache is marked stale and such inside of the updates for the BGP peers, without the update from the BGP config, then the secret cache will stop watching the BGP config secret. I originally had the update for the BGP config inside of the update BGP peers, but that seemed weirder to me than just pulling out the secret watcher logic.

Reply:
I think moving these is going to break normal BGP passwords though, isn't it?

The stale bit was being cleared by the calls to `c.setPeerConfigFieldsFromV3Resource` above, e.g. https://github.com/projectcalico/calico/pull/5609/files#diff-fe5c2e4b0c7489aaa0e63802827d4cb96860e24737e18a47051321f28ae74a17R619

Now that we don't mark / sweep surrounding those calls, won't potentially we mark a bunch of secrets as invalid and then sweep them without ever marking them as valid?

Reply:
Nevermind, I see now that `updatePeersV1` is what calls through to the above, so I think this is good.